spec/scenarios/dunning/campaign_v1_spec.rb

```diff
@@ -0,0 +1,232 @@
+          line2: nil,
+          postal_code: nil,
+          state: nil
+        },
+        email: "awdawd@desf.com",
+        name: "Testing Stripe",
+        phone: nil
+      },
+      card: {
+        brand: "visa",
+        checks: {
+          address_line1_check: nil,
+          address_postal_code_check: nil,
+          cvc_check: "pass"
+        },
+        country: "US",
+        display_brand: "visa",
+        exp_month: 12,
+        exp_year: 2028,
+        fingerprint: "8TOiB4cGytYxCweY",
+        funding: "credit",
+        generated_from: nil,
+        last4: "4242",
+        networks: {
+          available: [
+            "visa"
+          ],
+          preferred: nil
+        },
+        regulated_status: "unregulated",
+        three_d_secure_usage: {
+          supported: true
+        },
+        wallet: nil
+      },
+      created: 1741878064,
+      customer: stripe_cus_id,
+      livemode: false,
+      metadata: {},
+      type: "card"
+    }
+  end
+  let(:stripe_payment_intent_response) do
+    File.read("spec/fixtures/stripe/payment_intent_failed_card_declined.json")
+  end
+
+  def build_stripe_payment_intent_response
+    JSON.parse(stripe_payment_intent_response, symbolize_names: true).tap do |h|
+      h[:error][:payment_intent][:id] = "pi_#{SecureRandom.hex}"
+    end
+  end
+
+  # TODO: make it a test metadata `:scenarios, type: :request, premium: true`
+  around { |test| lago_premium!(&test) }
+
+  before do
+    stub_pdf_generation
+    stripe_customer
+    dunning_campaign_threshold
+
+    stub_request(:post, webhook_url).with do |req|
+      webhooks_sent << JSON.parse(req.body)
+      true
+    end.and_return(status: 200)
+
+    stub_request(:get, "https://api.stripe.com/v1/customers/#{stripe_customer.provider_customer_id}")
+      .and_return(status: 200, body: stripe_customer_response)
+    stub_request(:get, "https://api.stripe.com/v1/customers/#{stripe_customer.provider_customer_id}/payment_methods/pm_123456")
+      .and_return(status: 200, body: stripe_payment_method_response.to_json)
+    stub_request(:post, "https://api.stripe.com/v1/payment_intents")
+      .and_return(
+        status: 402,
+        body: lambda { |_req| build_stripe_payment_intent_response.to_json }
+      )
+    stub_request(:post, "https://api.stripe.com/v1/checkout/sessions")
+      .and_return(status: 200, body: {url: "https://stripe.com/checkout/session/cs_test_123"}.to_json)
+
+    WebMock.after_request do |request_signature, response|
+      if request_signature.uri.path.match?(%r{/v1/payment_intents})
+        request_body_hash = if request_signature.url_encoded?
+          Rack::Utils.parse_nested_query(request_signature.body)
+        elsif request_signature.body.json_encoded?
+          JSON.parse(request_signature.body)
+        end
+
+        Jobs::MockStripeWebhookEventJob.perform_later(
+          organization,
+          request_body_hash,
+          JSON.parse(response.body)
+        )
+      end
+    end
```

Comment:
Every time a payment intent is created on Stripe, we enqueue en job to simulate the Stripe webhook being received.
It's a new job in `spec/` so we can build mock some data and build the payload.